src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,190 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        validateIndex(index, size);
+        if (index == 0) {
+            addFirst(value);
+        } else if (index == size) {
+            addLast(value);
+        } else {
+            Node<T> oldNode = findElenemt(index);
+            Node<T> newNode = new Node<T>(oldNode.prev, value, oldNode);
+            oldNode.prev.next = newNode;
+            oldNode.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        validateIndex(index, size - 1);
+        return findElenemt(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T removed = remove(index);
+        add(value, index);
+        return removed;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        validateIndex(index, size - 1);
+        if (index == 0) {
+            return removeFirst();
+        } else if (index == size - 1) {
+            return removeLast();
+        } else {
+            Node<T> oldNode = findElenemt(index);
+            unlink(oldNode);
+            size--;
+            return oldNode.item;
+        }
     }
 
     @Override
     public boolean remove(T object) {
+        if (size != 0) {
+            int index = -1;
+            Node<T> currNode = head;
+            for (int i = 0; i < size; i++) {
+                if (currNode.item == object
+                        || (currNode.item != null && currNode.item.equals(object))) {
+                    index = i;
+                    break;
+                }
+                currNode = currNode.next;
```

Comment:
Also, you can try to use while loop, at least because you don't need an index here